docs/sops/serverless-controller-knative-up.md

```diff
@@ -0,0 +1,149 @@
+```
+
+3. Redeploy any failing controllers by restarting the deployments:
+
+```bash
+oc -n openshift-serverless rollout restart deployments/knative-openshift
+oc -n openshift-serverless rollout restart deployments/knative-openshift-ingress
+oc -n openshift-serverless rollout restart deployments/knative-operator-webhook
+```
+
+This should result in new pods getting deployed, attempt step (1) again and see if the pods achieve running state.
+
+5. Check to see if all the pods are running in `knative-serving` namespace:
+
+```bash
+$ oc -n knative-serving get pods -l app.kubernetes.io/name=knative-serving
+``` 
+
+6. If they are not running, look at the pod's logs/events to see what may be causing the issues. Please make sure to grab the logs/events so they can be shared with the engineering team later:
+
+```bash
+# Check pod logs 
+$ oc -n knative-serving logs -l app.kubernetes.io/name=knative-serving --prefix=true
+
+# Check events 
+$ oc -n knative-serving get events | grep pod
+
+# Check pod status fields
+$ oc -n knative-serving get pods -l app.kubernetes.io/name=knative-serving -o jsonpath="{range .items[*]}{.status}{\"\n\n\"}{end}"
+```
+
+7. Redeploy Knative Serving controllers by restarting the deployments:
+
+```bash
+$ oc -n knative-serving rollout restart deployments -l app.kubernetes.io/name=knative-serving
+```
+
+This should result in new pods getting deployed, attempt step (5) again and see if the pods achieve running state.
+
+8. If Kourier is used as the Knative Serving Ingress, check to see if all the relevant pods are running in `knative-serving-ingress` namespace:
+
+```bash
+$ oc -n knative-serving-ingress get pods -l 'app in (3scale-kourier-gateway, net-kourier-controller)'
+```
+
+9. If they are not running, look at the pod's logs/events to see what may be causing the issues. Please make sure to grab the logs/events so they can be shared with the engineering team later:
+
+```bash
+# Check pod logs 
+$ oc -n knative-serving-ingress logs -l 'app in (3scale-kourier-gateway, net-kourier-controller)' --prefix=true
+
+# Check events 
+$ oc -n knative-serving-ingress get events | grep pod
+
+# Check pod status fields
+$ oc -n knative-serving-ingress get pods -l 'app in (3scale-kourier-gateway, net-kourier-controller)' -o jsonpath="{range .items[*]}{.status}{\"\n\n\"}{end}"
+```
+
+10. Redeploy Knative Serving Ingress controllers by restarting the deployments:
+
+```bash
+$ oc -n knative-serving-ingress rollout restart deployments -l app.kubernetes.io/component=net-kourier
+```
+
+This should result in new pods getting deployed, attempt step (8) again and see if the pods achieve running state.
+
+11. If Istio is used as the Knative Serving Ingress, check Istio status. You may find the SOPs for Istio by contacting Istio support. 
+
+12. If the problem persists, capture the logs and escalate to OpenShift Serverless engineering team with a Knative ["must-gather"](https://github.com/openshift-knative/must-gather) dump.
```

Comment:
Could you add the command here:

```oc adm must-gather --image=registry.redhat.io/openshift-serverless-1/svls-must-gather-rhel8 ```

That is the image I see people use. Not sure if there is a difference compared to quay.io/openshift-knative/must-gather.

Also it would be useful to add the SM one.

Reply:
I think SM uses `registry.redhat.io/openshift-service-mesh/istio-must-gather-rhel8:2.4.0` but needs verification.

Reply:
Removed the mention to must gather.

Reply:
They could also reach support or some contact?

Reply:
Good feedback, thanks. Let me find some links to put in here :)

Reply:
@skonto added some links about how to reach out to SM team.
(I've asked in Istio team channel)